padertorch/train/hooks.py

```diff
@@ -946,8 +946,13 @@ def uid(self):
     def get_optimizer(self, trainer):
         optimizer = trainer.optimizer
         if self.name is not None:
-            assert isinstance(optimizer, dict)
+            assert isinstance(optimizer, dict), type(optimizer)
             optimizer = optimizer[self.name]
+        else:
+            assert (
+                hasattr(optimizer, 'optimizer')
+                and hasattr(optimizer.optimizer, 'param_groups')
+            ), type(optimizer)
```

Comment:
This assert can as well be performed when `self.name is not None:`, the same assertion should hold for the contents of the optimizer dict. Can you remove the `else:`?